.github/workflows/test.yml

```diff
@@ -29,42 +29,29 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        python-version:
-        - "3.9"
-        - "3.10"
-        - "3.11"
-        - "3.12"
-        - "3.13"
-        os:
-        - "windows-latest"
-        - "macos-latest"
-        - "ubuntu-latest"
-        # Python 3.9 is on macos-13 but not macos-latest (macos-14-arm64)
-        # https://github.com/actions/setup-python/issues/696#issuecomment-1637587760
-        exclude:
-        - { python-version: "3.9", os: "macos-latest" }
-        include:
-        - { python-version: "3.9", os: "macos-13" }
-
+        python-version: ["3.9", "3.10", "3.11", "3.12", "3.13"]
+        os: [windows-latest, macos-latest, ubuntu-latest]
 
     steps:
       - uses: actions/checkout@v4
+        with:
+          persist-credentials: false
+
       - name: Set up Python ${{ matrix.python-version }}
         uses: actions/setup-python@v5
         with:
           python-version: ${{ matrix.python-version }}
-          cache: pip
           allow-prereleases: true
 
-      - name: Install dependencies
-        run: |
-          python -m pip install -U pip
-          python -m pip install -U wheel
-          python -m pip install -U tox
+      - name: Install uv
+        uses: hynek/setup-cached-uv@v2
+        with:
+          cache-dependency-path: |
+            requirements.txt
 
-      - name: Run tests
+      - name: Tox tests
         run: |
-          tox -e py -- -v --cov-report term
+          uvx --with tox-uv tox -e py -- -v --cov-report term
```

Comment:
I prefer having an explicit "install" step, but appreciate that this is the style that you've used in several other repos now, so might be worth it from the consistency perspective.